src/utils/object.js

```diff
@@ -44,3 +46,37 @@ export function map(opt_initial) {
 export function hasOwn(obj, key) {
   return hasOwn_.call(obj, key);
 }
+
+/**
+ * Deep merge object b into object a. Both a and b can only contain
+ * primitives, arrays, and plain objects. For any conflicts, object b wins.
+ * Arrays are replaced, not merged. Plain objects are recursively merged.
+ * @param {!Object} a the destination object
+ * @param {!Object} b
+ * @param {number=} opt_maxDepth The maximum depth for deep merge, beyond which
+ *                               Object.assign will be used.
+ * @param {number=} opt_depth The current depth of the objects being merged.
+ * @return {!Object}
+ */
+export function deepMerge(a, b, opt_maxDepth, opt_depth) {
```

Comment:
Use a private helper method `deepMerge_()` to hide the current depth param. No need to expose that to outside of this class.